lib/dialects/postgres/data-types.js

```diff
@@ -34,9 +34,44 @@ module.exports = BaseTypes => {
   inherits(DATEONLY, BaseTypes.DATEONLY);
 
   DATEONLY.parse = function parse(value) {
+    if (value === 'infinity') {
+      value = Infinity;
+    } else if (value === '-infinity') {
+      value = -Infinity;
+    }
+
+    return value;
+  };
+
+  DATEONLY.prototype._stringify = function _stringify(value, options) {
+    if (value === Infinity) {
+      return 'Infinity';
+    } else if (value === -Infinity) {
+      return '-Infinity';
+    }
+
+    return BaseTypes.DATEONLY.prototype._stringify.call(this, value, options);
+  };
+
+  DATEONLY.prototype._sanitize = function _sanitize(value, options) {
+    if ((!options || options && !options.raw) && value !== Infinity && value !== -Infinity) {
+      value = BaseTypes.DATEONLY.prototype._sanitize.call(this, value);
+    }
+
     return value;
   };
 
+  DATEONLY.prototype._isChanged = function _isChanged(value, originalValue) {
+    if (
+      originalValue && !!value &&
+      (value === originalValue || value instanceof Date && originalValue instanceof Date && value.getTime() === originalValue.getTime())
```

Comment:
move this to base type? this logic should be honored by all dialects for DATEONLY